File: src/main/java/com/linebot/client/self/SelfClient.java
package com.linebot.client.self;

import com.linebot.client.WebClient;
import com.linebot.config.PropertiesConfig;
import com.linebot.model.response.RestApiResponse;
import lombok.AllArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.stereotype.Component;

@Slf4j
@AllArgsConstructor
@Component
public class SelfClient {
    private WebClient webClient;
    private PropertiesConfig config;

    public RestApiResponse keepAlive() {
        String url = config.getOwnHost() + ":" + config.getOwnPort() + "/self/keep";
        return webClient.get(url, RestApiResponse.class);
    }
}

File: src/main/kotlin/com/linebot/client/self/SelfClient.kt
package com.linebot.client.self

import com.linebot.client.WebClient
import com.linebot.config.PropertiesConfig
import com.linebot.model.response.RestApiResponse
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Component

@Component
class SelfClient(
    private val webClient: WebClient,
    private val config: PropertiesConfig
) {
    val log: Logger = LoggerFactory.getLogger(SelfClient::class.java)

    fun keepAlive(): RestApiResponse {
        val url = "${config.ownHost}:${config.ownPort}/self/keep"
        return webClient[url, RestApiResponse::class.java] ?: RestApiResponse("error")
    }
}
